src/main/java/edu/ksu/canvas/attendance/submitter/AssignmentValidator.java

```diff
@@ -0,0 +1,89 @@
+
+    private static final Logger LOG = Logger.getLogger(AssignmentValidator.class);
+
+    public Error validateAttendanceAssignment(Long courseId, AttendanceAssignment attendanceAssignment, CanvasApiWrapperService canvasApiWrapperService, OauthToken oauthToken) {
+
+        // If there is no linked canvas assignment, creates on in canvas
+        if(attendanceAssignment.getCanvasAssignmentId() == null) {
+            return new Error("NO CANVAS ASSIGNMENT LINKED");
+        }
+
+        // Looks for the linked assignment in canvas
+        Optional<Assignment> assignmentOptional;
+        try {
+            assignmentOptional = canvasApiWrapperService.getSingleAssignment(courseId, oauthToken, attendanceAssignment.getCanvasAssignmentId().toString());
+        } catch (IOException e) {
+            LOG.error("Error while getting assignment from canvas for section: " + attendanceAssignment.getAttendanceSection().getSectionId(), e);
+            return new Error("Could not connect to Canvas to get assignment");
+        } catch (ObjectNotFoundException e) {
+            LOG.debug("Assignment with ID: " + attendanceAssignment.getCanvasAssignmentId() + "not found in Canvas.");
+            return new Error("NO CANVAS ASSIGNMENT LINKED");
+        }
+
+        if(!assignmentOptional.isPresent()) {
+            return new Error("NO CANVAS ASSIGNMENT LINKED");
+        }
+
+        return null;
+    }
+
+    public Error validateCanvasAssignment(CourseConfigurationForm courseConfigurationForm, Long courseId, AttendanceAssignment attendanceAssignment, CanvasApiWrapperService canvasApiWrapperService, OauthToken oauthToken) {
+
+        //Look for changes between the course configuration in the form and in the DB
+        if(!compareFormToDB(courseConfigurationForm, attendanceAssignment)) {
+            LOG.debug("Configuration form is different than saved assignment configuration for section: " + attendanceAssignment.getAttendanceSection().getSectionId());
+            return new Error("Assignment configuration needs to be saved before pushing to Canvas");
+        }
+
+        //Look for the canvas assignment in canvas
+        Optional<Assignment> assignmentOptional;
+        try {
+            assignmentOptional = canvasApiWrapperService.getSingleAssignment(courseId, oauthToken, attendanceAssignment.getCanvasAssignmentId().toString());
+        } catch (IOException e) {
+            LOG.error("Error while getting assignment from canvas for section: " + attendanceAssignment.getAttendanceSection().getSectionId(), e);
+            return new Error("Could not connect to Canvas to get assignment");
+        }
+
+        if(!assignmentOptional.isPresent()) {
+            return new Error("Assignment not found in Canvas");
+        }
+
+        //Looks for discrepancy between the assignment in canvas and in database
+        if (assignmentOptional.get().getPointsPossible().doubleValue() != courseConfigurationForm.getAssignmentPoints().doubleValue()) {
```

Comment:
Nested gets like this can cause NPEs if the object is not validated. In this case, getPointsPossible() and getAssignmentsPoints() could be null. It is especially dangerous in the former case because the object was just retrieved from a library (which we control but currently make no guarantees about lack of nulls) which gets the data from an external web service (which we do not control and could change)